modelkit/core/model.py

```diff
@@ -35,6 +36,34 @@
 
 logger = get_logger(__name__)
 
+ModelDependency = TypeVar(
+    "ModelDependency", bound=Union["Model", "AsyncModel", "WrappedAsyncModel"]
+)
+
+
+class ModelDependenciesMapping:
+    def __init__(
+        self, models: Dict[str, Union["Model", "AsyncModel", "WrappedAsyncModel"]]
+    ) -> None:
+        self.models = models
+
+    def __getitem__(self, key):
+        return self.models[key]
+
```

Comment:
The mock here is meant to be used in read-only inside the `.predict` methods. 
I believe we really want to avoid users updating the model dependencies. 
Any dictionary fed to a `Model` will be turned into a `ModelDependenciesMapping` at initialization so one can still just do it outside of the `Model` class